### PR TITLE
setup a matrix build

### PR DESCRIPTION
**Pull request by unglaublicherdude**

### CODE REVIEW

File: get-matrix.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+MIN_VERSION=$([[ $(cat appinfo/info.xml) =~ $(echo 'min-version=[^[0-9]]*([0-9]+).*') ]] && echo ${BASH_REMATCH[1]})
+MAX_VERSION=$([[ $(cat appinfo/info.xml) =~ $(echo 'max-version=[^[0-9]]*([0-9]+).*') ]] && echo ${BASH_REMATCH[1]})
+
+BASE_TEMPLATE='[VERSIONS]'
+VERSION_TEMPLATE='"NEXTCLOUD_VERSION"'
+
+if [ -z "$MIN_VERSION" ] && [ -z "$MAX_VERSION" ]; then
+    echo '["29"]'
+    exit 0
```
Comment (GermanCoding) [inline get-matrix.sh:11]: I don't like having "the current version" defined in too many places - that gets annoying on upgrades and I personally tend to forget to update this stuff. Can't we just error here? Either there's something defined in info.xml or something is broken?

File: get-matrix.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+FIRST_RUN="YES"
+for i in $(eval echo {$MIN_VERSION..$MAX_VERSION})
+do
+    if [ "$FIRST_RUN" == "YES" ]; then
+        VERSIONS="$(echo $VERSION_TEMPLATE | sed "s/NEXTCLOUD_VERSION/$i/g")"
+        FIRST_RUN="NO"
+        continue
+    fi
+    VERSIONS="$VERSIONS,$(echo $VERSION_TEMPLATE | sed "s/NEXTCLOUD_VERSION/$i/g")"
+done
```
Comment (GermanCoding) [inline get-matrix.sh:46]: ChatG... *hust hust ahem* I came up with this idea to get rid of the divergent control flow:

```suggestion
VERSIONS=""
DELIMITER=""
for i in $(eval echo {$MIN_VERSION..$MAX_VERSION})
do
    VERSIONS="${VERSIONS}${DELIMITER}$(echo $VERSION_TEMPLATE | sed "s/NEXTCLOUD_VERSION/$i/g")"
    DELIMITER=","
done
```

It's definitely shorter, but I'm not entirely convinced it's significantly better for readability. WDYT?

**Review by GermanCoding — COMMENTED** [review]
Very nice, love the idea!

File: get-matrix.sh (hunk 2 above)
Comment (unglaublicherdude) [reply get-matrix.sh:46]: I like it tbh. its so simple 😄

File: get-matrix.sh (hunk 1 above)
Comment (unglaublicherdude) [reply get-matrix.sh:11]: Yeah you are right. I might be better to get notified when the constraints are empty.

**Comment by unglaublicherdude** [comment]
Looking at a füll blown info.xml I should also adjust the reges to include <nextcloud> to be sure we are looking at the right constraints.

https://docs.nextcloud.com/server/latest/developer_manual/app_development/info.html

**Comment by unglaublicherdude** [comment]
I refactored the script regarding to the mentioned issues. It is a much shorter version now.

**Pull request merged**